rosbag2_transport/test/rosbag2_transport/test_record_all_no_discovery.cpp

```diff
@@ -24,19 +24,22 @@
 
 TEST_F(RecordIntegrationTestFixture, record_all_without_discovery_ignores_later_announced_topics)
 {
+  auto topic = "/string_topic";
   auto string_message = get_messages_strings()[0];
   string_message->string_value = "Hello World";
 
   start_recording({true, true, {}, "rmw_format", 1ms});
 
   std::this_thread::sleep_for(100ms);
   auto publisher_node = std::make_shared<rclcpp::Node>("publisher_for_test");
-  auto publisher = publisher_node->create_publisher<test_msgs::msg::Strings>("/string_topic", 10);
+  auto publisher = publisher_node->create_publisher<test_msgs::msg::Strings>(topic, 10);
   for (int i = 0; i < 5; ++i) {
     std::this_thread::sleep_for(20ms);
     publisher->publish(*string_message);
   }
   stop_recording();
 
-  ASSERT_THAT(writer_->get_messages(), SizeIs(0));
+  for (auto & msg : writer_->get_messages()) {
+    EXPECT_NE(msg->topic_name, topic);
+  }
```

Comment:
I think it's easier to replace this for loop with a call to `writer_->get_topics()`.
Something like:

```
auto recorded_topics = writer_->get_topics();
EXPECT_EQ(0u, recorded_topics.count(topic));
```

Reply:
Good idea. Will do.